src/libs/SessionUtils.js

```diff
@@ -0,0 +1,29 @@
+import lodashGet from 'lodash/get';
+
+/**
+ * Determine if the transitioning user is logging in as a new user.
+ *
+ * @param {String} transitionURL
+ * @param {String} sessionEmail
+ * @returns {Boolean}
+ */
+function isLoggingInAsNewUser(transitionURL, sessionEmail) {
+    // The OldDot mobile app does not URL encode the parameters, but OldDot web
+    // does. We don't want to deploy OldDot mobile again, so as a work around we
+    // compare both the raw and decoded email from the transition link. If both don't match
+    // the session email, then we are logging in as a new user.
+    // See https://github.com/Expensify/Mobile-Expensify/pull/12280#issuecomment-1155633503
+    const params = new URLSearchParams(transitionURL);
+    const paramsEmail = params.get('email');
+
+    // Capture the un-encoded text in the email param
+    const emailParamRegex = /[?&]email=([^&]*)/g;
+    const matches = emailParamRegex.exec(transitionURL);
+    const linkedEmail = lodashGet(matches, 1, null);
+    return paramsEmail !== sessionEmail && linkedEmail !== sessionEmail;
```

Comment:
Should this be `&&` or `||`?

Reply:
I think it is right with `&&`. He wants to return true only if both params are different from session email.

Reply:
Why?

Reply:
Yes, what @danieldoglas said. It could be converted to the following, but I think what I have is more clear.
`return !(paramsEmail === sessionEmail || linkedEmail === sessionEmail)`

We want to determine if we are transitioning the new user or not. If the emails match then we have already signed in the correct user and we can navigate to the exit route.

If the params email has been decoded from an un-encoded email passed from mobile `isLogginInAsNewUser` will return false.
```
'neil test@expensifail.com' !== 'neil+test@expensifail.com' && 'neil+test@expensifail.com' !== 'neil+test@expensifail.com'`
```

Any more questions?

Reply:
Discussed with @neil-marcellini in person, but I think we just need to validate that all four cases work with this logic:
1. Unencoded new user (OldApp new user)
2. Encoded new user (OldDot new user)
3. Unencoded existing user (OldApp existing user)
4. Encoded existing user (OldDot existing user)

Reply:
Andrew brought up the fact that this fails on web when you are not logging in as a new user, so I updated it to return early and all 4 cases make sense now.